app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,63 @@
+class Admin::UsersController < ApplicationController
+  before_action :admin_user, only: [:new, :index, :show, :edit, :update, :destroy]
+
+PER = 4
+
+  def index
+    @users = User.all.order(created_at:"DESC")
+    @users = @users.page(params[:page]).per(PER)
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @tasks = @user.tasks
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to admin_users_path, notice:"#{@user.name}を登録しました」"
+    else
+      render :new
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to admin_user_path(@user), notice:"「#{@user.name}」を更新しました"
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @user = User.find(params[:id])
+    if @user.destroy
+      redirect_to admin_users_path, notice:"「#{@user.name}」削除"
+    else
+      redirect_to admin_users_path, notice:"「#{@user.name}」削除"
+    end
```

Comment:
削除できないケースはほとんど発生しないかと思うので、elseで書かなくても破壊的メソッドとしてエラーを出しておく程度でも良いかもしれません。
```suggestion
    if @user.destroy!
      redirect_to admin_users_path, notice:"「#{@user.name}」削除"
    end
```